runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/DoFnInfo.java

```diff
@@ -44,10 +44,20 @@
   private final Map<Long, TupleTag<?>> outputMap;
 
   /**
-   * Creates a {@link DoFnInfo} for the given {@link DoFn} or {@link OldDoFn} and auxiliary bits and
-   * pieces.
+   * Creates a {@link DoFnInfo} for the given {@link Serializable} object, which is expected to be a
+   * {@link DoFn} or {@link OldDoFn} or other context-appropriate UDF blob.
    */
-  public DoFnInfo(
+  public static <InputT, OutputT> DoFnInfo<InputT, OutputT> forFn(
+      Serializable doFn,
+      WindowingStrategy<?, ?> windowingStrategy,
+      Iterable<PCollectionView<?>> sideInputViews,
+      Coder<InputT> inputCoder,
+      long mainOutput,
+      Map<Long, TupleTag<?>> outputMap) {
+    return new DoFnInfo(doFn, windowingStrategy, sideInputViews, inputCoder, mainOutput, outputMap);
+  }
+
+  private DoFnInfo(
```

Comment:
nit: The comment below on the deprecated constructor says to now use the private DoFnInfo instead of the method forFn